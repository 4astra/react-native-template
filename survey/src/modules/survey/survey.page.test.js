import React from 'react';
import SurveyPage from '../survey/survey.page';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

const case1 = {
  surveyReducer: {
    loading: false,
    result: [],
  },
  homeReducer: {
    selected: {
      "name": "Melun",
      "code": "XX1"
    }
  }
};

const case2 = {
  surveyReducer: {
    loading: false,
    result: [],
  },
  homeReducer: {
    selected: {
      "name": "Melun",
      "code": null
    }
  }
};

describe('SurveyPage', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <SurveyPage />,
      { context: { store: mockStore(case1) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders as expected when code is null', () => {
    const wrapper = shallow(
      <SurveyPage />,
      { context: { store: mockStore(case2) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});

