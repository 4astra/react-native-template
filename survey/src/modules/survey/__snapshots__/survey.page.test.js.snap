// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyPage renders as expected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
      }
    }
  >
    <FlatList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "marginTop": 15,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        style={
          Object {
            "marginTop": 15,
          }
        }
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "_element": <SurveyPage
      actions={
        Object {
          "default": [Function],
          "survey": [Function],
        }
      }
      info={
        Object {
          "code": "XX1",
          "name": "Melun",
        }
      }
      result={Array []}
    />,
    "_instance": SurveyPage {
      "context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "props": Object {
        "actions": Object {
          "default": [Function],
          "survey": [Function],
        },
        "info": Object {
          "code": "XX1",
          "name": "Melun",
        },
        "result": Array [],
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        style={
          Object {
            "marginTop": 15,
          }
        }
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SurveyPage
    actions={
      Object {
        "default": [Function],
        "survey": [Function],
      }
    }
    info={
      Object {
        "code": "XX1",
        "name": "Melun",
      }
    }
    result={Array []}
  />,
}
`;

exports[`SurveyPage renders as expected when code is null 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
      }
    }
  >
    <FlatList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "marginTop": 15,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        style={
          Object {
            "marginTop": 15,
          }
        }
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "_element": <SurveyPage
      actions={
        Object {
          "default": [Function],
          "survey": [Function],
        }
      }
      info={
        Object {
          "code": null,
          "name": "Melun",
        }
      }
      result={Array []}
    />,
    "_instance": SurveyPage {
      "context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "props": Object {
        "actions": Object {
          "default": [Function],
          "survey": [Function],
        },
        "info": Object {
          "code": null,
          "name": "Melun",
        },
        "result": Array [],
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        style={
          Object {
            "marginTop": 15,
          }
        }
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SurveyPage
    actions={
      Object {
        "default": [Function],
        "survey": [Function],
      }
    }
    info={
      Object {
        "code": null,
        "name": "Melun",
      }
    }
    result={Array []}
  />,
}
`;
